src/rpdk/core/contract/resource_client.py

```diff
@@ -69,6 +69,24 @@ def prune_properties_if_not_exist_in_path(output_model, input_model, paths):
     return output_document["properties"]
 
 
+def prune_properties_which_dont_exist_in_path(model, paths):
+    """Prune given properties from a model. This method removes any property
```

Comment:
"Prunes model to properties present in path." seems easier to understand to me than "Prune given properties from a model."